pyfixest/estimation/feols_.py

```diff
@@ -478,7 +478,7 @@ def vcov(
         # update p-value, t-stat, standard error, confint
         self.get_inference()
 
-        return self
+        return self._vcov
```

Comment:
I just realized that things might be a little bit more complicated than just this simple change

Doing as above would introduce a breaking changes as users could no longer run code as 

```py
import pyfixest as pf
pf.feols("Y~X1", data = pf.get_data).vcov("hetero").summary()
```

I'm not really sure if many use such syntax, but I have certainly seen it already =) 

One option would be to add a "return_vcov" function argument, which defaults to False, in which case the `Feols` object would be returned (i.e. the status quo), whereas if `True`, we'd return the full vcov matrix. 

I also wonder if we should add a function that allows users to control if they want to overwrite inferential results in the Feols instance whenever they call `.vcov`? 

I.e. right now, whenever `vcov()` is called, the `_vcov` attribute of the Feols instance is updated, and `self.get_inference()` updates t-statistics, p-values, etc. 

What do you think @greenguy33 ?